docs/DeveloperGuide.md

```diff
@@ -58,20 +58,49 @@ The StorageHandler interacts with the other classes as shown in the *Sequence Di
 where it shows how the StorageHandler loads the local user data json file as well as the user plans json file upon the 
 resumption of the program.
 <div align="center">
-<img src="UML/Images/LoadingUserCareerData.png" />
+<img src="UML/Images/LoadingUserCareerData.png"/>
 <p>
 Figure X.2
 </p>
 </div>
 
+### Command Handler Component
+
+The command handler component consists of multiple states that the program can be at any time. 
+It controls where the user input is being processed.
+
+#### Key Aspects: 
+
+* Handles when there is an exercise ongoing, giving access to finishing and hence saving finished exercises, and keeping track of what exercises there are.
+* Handles when there is no exercise ongoing, granting access to generating new exercises with different confines.
+
+<img src="UML/Images/CommandHandler.png" />
+
+### ExerciseSessionCommandHandler
+The exercise Command Handler is engaged when the user has an ongoing workout.
+Upon completion of the workout, the user can save their workout if completed, such
+that the completed workout is logged and will be saved for future reference.
+
+To manage this exists the ```ExerciseStateHandler```, which allows for saving, starting
+ending, cancelling workouts.
+
+<img src="UML/Images/CommandHandler.png" />
+

```

Comment:
CommandHandler and the other handlers I think need to make it less coupled and into like an interface as per AB3